src/main/resources/jshint-shell.js

```diff
@@ -28,6 +28,13 @@
     var OPTIONS_ARG = 4;
 
     var options = JSON.parse(stripJsonComments(args[OPTIONS_ARG]));
+    var globals;
+
+    if (options.globals) {
+        globals = options.globals;
+        delete options.globals;
```

Comment:
I believe its just assigning a reference to `options.globals`.... but somehow JavaScript knows how to manage that memory even if `options` no longer has a `globals` property.   I'm not sure that my explanation is accurate.  

I think what matters here is that `options` no longer has a `globals` property
```
$ node
> var options = {undef: true, globals: {"$": true}};
undefined
> var globals = options.globals;
undefined
> delete options.globals;
true
> globals
{ '$': true }
> options.globals
undefined
> options
{ undef: true }
```

whats funny is that `globals` is not a copy. as you say.
```
> var options = {undef: true, globals: {"$": true}};
undefined
> var globals = options.globals;
undefined
> options.globals.define = true
true
> globals
{ '$': true, define: true }
```